src/main/java/genie/task/Task.java

```diff
@@ -1,30 +1,63 @@
 package genie.task;
 
+/**
+ * Deals with creating tasks.
+ */
 public class Task {
     protected String description;
     protected boolean isDone;
 
+    /**
+     * A constructor that takes in the task descriptor and initialises task as undone.
+     * @param description
+     */
     public Task(String description) {
         this.description = description;
         this.isDone = false;
     }
 
+    /**
+     * Retrieves the status of the task (marked done or undone).
+     * @return "X" if marked done, " " otherwise
+     */
     public String getStatusIcon() {
         return (isDone ? "X" : " "); // mark done task with X
     }
+
+    /**
+     * Changes the status of the task to marked done.
+     */
     public void markDone() {
         this.isDone = true;
     }
+    /**
+     * Changes the status of the task to marked undone.
+     */
     public void unmarkDone() {
```

Comment:
Just a small nit, but would "markUndone" sound more natural and easier to remember?